src/AWSLambda/Events/APIGateway.hs

```diff
@@ -132,7 +134,20 @@ data APIGatewayProxyRequest body = APIGatewayProxyRequest
   , _agprqStageVariables        :: !(HashMap StageVarName StageVarValue)
   , _agprqRequestContext        :: !ProxyRequestContext
   , _agprqBody                  :: !(Maybe (TextValue body))
-  } deriving (Eq, Show, Generic)
+  } deriving (Show, Generic)
+
+instance Eq body => Eq (APIGatewayProxyRequest body) where
+  (==) =
+    (==) `on` \rq ->
+      ( _agprqResource rq
+      , _agprqPath rq
+      , _agprqHttpMethod rq
+      , Set.fromList (_agprqHeaders rq)
```

Comment:
Perhaps add a comment on why this is necessary?

Reply:
Is it just to make the unit tests easier to write? If so, maybe it would be better to just change the test to not use `Eq`.